src/Controller/AwsStorage.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Controller;
+
+use App\Command\CommandDispatcherTrait;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use League\Flysystem\AwsS3v3\AwsS3Adapter;
+use League\Flysystem\Filesystem;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Aws\Credentials\CredentialProvider;
+
+class AwsStorage extends AbstractController
+{
+    use CommandDispatcherTrait;
+
+    /**
+     * @Route("/cfdoc/aws", name="aws_storage_file")
+     */
+    public function awsStorage(Request $request)
+    {
+        $form = $this->createFormBuilder()
+        ->add('upload', FileType::class, array('label' => 'File Upload'))
+        ->add('save', SubmitType::class, array('label' => 'Submit'))
+        ->getForm();
+
+        $form->handleRequest($request);
+        $filesystem = $this->configuration();
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            $fileName = $data['upload']->getClientOriginalName();
+            $file = $data['upload']->getRealPath();
+            if ($filesystem->has($fileName))
+            {echo '<div class="message">Already exists</div>';}
+            else
+            {
+                $stream = fopen($file, 'r+');
+                $result =$filesystem->writeStream($fileName, $stream);
+                fclose($stream);
+                echo '<div class="message">File Uploaded Successfully..!!</div>';
+            }
+        }
+       $aws_data = $filesystem->listContents();
+        return $this->render('framework/doc_tree/aws_upload.html.twig', array(
+            'form' => $form->createView(),
+            'aws_files' => $aws_data,
+        ));
+    }
+
+    private function configuration()
```

Comment:
Maybe this can be moved to a service class in the `src/Service` dir. @roverwolf what do you think?

Reply:
Yes, the configuration should be a service, possibly just using passed environment variables and defining services in the services.yml file, or if more complicated then using services to do the configuration.

Since this is also the *only* place that specifies that this is using S3 we might want to rename the classes to change the Aws portion and allow for the possibility of using other adapters as well, with S3 being the only supported adapter (or maybe allow the local adapter to make it easier for developers?  or developers could use Minio for testing with?).